Content.Server/Guardian/GuardianSystem.cs

```diff
@@ -0,0 +1,199 @@
+using Content.Server.Actions.Actions;
+using Content.Server.Lathe.Components;
+using Content.Server.Popups;
+using Content.Shared.Actions;
+using Content.Shared.Actions.Components;
+using Content.Shared.Audio;
+using Content.Shared.Damage;
+using Content.Shared.Examine;
+using Content.Shared.Interaction;
+using Content.Shared.MobState;
+using Content.Shared.MobState.Components;
+using Content.Shared.MobState.EntitySystems;
+using Content.Shared.Popups;
+using JetBrains.Annotations;
+using Robust.Shared.Audio;
+using Robust.Shared.Containers;
+using Robust.Shared.GameObjects;
+using Robust.Shared.Localization;
+using Robust.Shared.Log;
+using Robust.Shared.Map;
+using Robust.Shared.Player;
+using System;
+using System.Timers;
+
+namespace Content.Server.Guardian
+{
+    public class GuardianSystem : EntitySystem
+    {
+        public override void Initialize()
+        {
+            base.Initialize();
+            SubscribeLocalEvent<GuardianCreatorComponent, UseInHandEvent>(OnGuardianCreated);
+            SubscribeLocalEvent<GuardianCreatorComponent, ExaminedEvent>(OnActivatorExamined);
+            SubscribeLocalEvent<GuardianComponent, MoveEvent>(OnGuardianMove);
+            SubscribeLocalEvent<GuardianHostComponent, MoveEvent>(OnGuardianHostMove);
+            SubscribeLocalEvent<GuardianComponent, DamageChangedEvent>(OnGuardianDamaged);
+            SubscribeLocalEvent<GuardianHostComponent, MobStateChangedEvent>(OnHostStateChange);
+        }
+
+        /// <summary>
+        /// Triggers when the host recives damage which puts in either critical or killed state
+        /// </summary>
+        /// <param name="uid"></param>
+        /// <param name="component"></param>
+        /// <param name="args"></param>
+        private void OnHostStateChange(EntityUid uid, GuardianHostComponent component, MobStateChangedEvent args)
+        {
+            if (EntityManager.GetEntity(component._hostedguardian) != null)
+            {
+                if (args.State.IsCritical())
+                {
+                    var guard = EntityManager.GetEntity(component._hostedguardian);
+                    guard.PopupMessage(Loc.GetString("guardian-host-critical-warn"));
+                    SoundSystem.Play(Filter.Local(), "/Audio/Effects/guardian_warn.ogg", uid);
+                }
+                else if (args.State.IsDead())
+                {
+                    var guard = EntityManager.GetEntity(component._hostedguardian);
+                    SoundSystem.Play(Filter.Pvs(guard), "/Audio/Voice/Human/malescream_guardian.ogg", uid, AudioHelpers.WithVariation(0.25f));
+                    EntityManager.QueueDeleteEntity(guard);
+                    EntityManager.QueueDeleteEntity(uid);
+                }
+            } 
+        }
+
```

Comment:
1. Don't use GetEntity unless you really need it (you don't), get components.
2. Use PopupSystem
3. Should the owner really get deleted if they die?

Reply:
Yes. The idea is that it's no less then a snap out of existence. I believe we discussed it and for now it's valid, maybe when proper disintegration is added this can be tweaked.

All else will be fixed